google/ads/google_ads/interceptors/logging_interceptor.py

```diff
@@ -117,7 +117,10 @@ def _get_customer_id(self, request):
         Args:
             request: An instance of a request proto message.
         """
-        return getattr(request, 'customer_id', None)
+        if hasattr(request, 'customer_id'):
+            return getattr(request, 'customer_id')
+        elif hasattr(request, 'resource_name'):
+            return getattr(request, 'resource_name').split('/')[-1]
```

Comment:
This needs an additional check to see if the resource_name indeed starts with customer_id.

This logic will fail for languageConstants/{criterionId}, for instance,
Perhaps the Ruby code also needs some fix.

Reply:
You can refer to https://github.com/googleads/google-ads-dotnet/blob/master/src/V3/ResourceNames.cs, which has a manually curated list of all the known resource names.

Reply:
@AnashOommen the issue isn't really about the structure of the resource name, right? It's that the customer ID happens to be in a resource name on a ` GetCustomerRequest` instance? Maybe it's better to check the type of the request object instead of parsing the resource name string? Do we know if there are any other request objects that don't have a `customer_id` field?

Reply:
Talked with @AnashOommen offline, the idea here is just to do some light validation of the string to make sure it starts with `'customers'` so we can be sure it's a resource_name, otherwise return `None`. No need for a complete list, also type checking doesn't quite make sense because an arbitrary string can be set by the user on `GetCustomerRequest`. I'll push up a change.